File: templates/problems/18_bit_manipulation/06_sum_two_integers/go/sum_two_integers_test.go
package sum_two_integers

import "testing"

func TestGetSum(t *testing.T) {
	tests := []struct {
		name     string
		a        int
		b        int
		expected int
	}{
		{"both positive", 1, 2, 3},
		{"positive and negative", 2, -1, 1},
		{"both negative", -1, -1, -2},
		{"zero and number", 0, 5, 5},
		{"both zero", 0, 0, 0},
		{"larger numbers", 100, 200, 300},
		{"negative result", -5, 3, -2},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := getSum(tt.a, tt.b)
			if got != tt.expected {
				t.Errorf("getSum(%d, %d) = %d, want %d", tt.a, tt.b, got, tt.expected)
			}
		})
	}
}

File: templates/problems/18_bit_manipulation/06_sum_two_integers/typescript/sumTwoIntegers.test.ts
import { describe, it, expect } from 'vitest';
import { getSum } from './sumTwoIntegers';

describe('SumTwoIntegers', () => {
    it('both positive', () => {
        expect(getSum(1, 2)).toBe(3);
    });

    it('positive and negative', () => {
        expect(getSum(2, -1)).toBe(1);
    });

    it('both negative', () => {
        expect(getSum(-1, -1)).toBe(-2);
    });

    it('zero and number', () => {
        expect(getSum(0, 5)).toBe(5);
    });

    it('both zero', () => {
        expect(getSum(0, 0)).toBe(0);
    });

    it('larger numbers', () => {
        expect(getSum(100, 200)).toBe(300);
    });

    it('negative result', () => {
        expect(getSum(-5, 3)).toBe(-2);
    });
});
